backend/src/main/java/pl/edu/agh/awiteks_backend/services/ForumService.java

```diff
@@ -83,35 +86,70 @@ private void removeForumThreadFromUserList(int id) {
+                this.forumRepository.findById(threadId).orElseThrow();
         final User creator = userRepository.findById(authorId).orElseThrow();
-        final ForumPost post = new ForumPost(creator, thread, postRequestBody.content());
+        final ForumPost post =
+                new ForumPost(creator, thread, postRequestBody.content());
         thread.addForumPost(post);
         creator.getForumPostList().add(post);
         this.postRepository.save(post);
         return post;
     }
 
-    public List<ForumPost> getPostsFromThread(Integer threadId) {
-        final ForumThread thread = this.forumRepository.findById(threadId).orElseThrow();
+    public List<ForumPostUserIncluded> getPostsUserIncludedFromThread(
+            int threadId) {
+        return getPostsFromThread(threadId).stream()
+                .map(ForumMapper::mapForumPostToForumPostUserIncluded).collect(
+                        Collectors.toList());
+    }
+
+    private List<ForumPost> getPostsFromThread(Integer threadId) {
+        final ForumThread thread =
+                this.forumRepository.findById(threadId).orElseThrow();
         return thread.getForumPosts();
     }
 
-    public ForumPost editPost(Integer threadId, Integer postId, AddPostRequestBody postRequestBody, Integer userId){
-        final ForumThread thread = this.forumRepository.findById(threadId).orElseThrow();
+    public ForumPost editPost(Integer threadId, Integer postId,
+                              AddPostRequestBody postRequestBody,
+                              Integer userId) {
+        final ForumThread thread =
+                this.forumRepository.findById(threadId).orElseThrow();
         final User creator = userRepository.findById(userId).orElseThrow();
-        final ForumPost post = this.postRepository.findById(postId).orElseThrow();
-        if(!(post.getUser() == creator)) throw new IllegalCallerException();
+        final ForumPost post =
+                this.postRepository.findById(postId).orElseThrow();
+        if (!(post.getUser() == creator)) {
+            throw new IllegalCallerException();
+        }
         creator.getForumPostList().remove(post);
         post.setContent(postRequestBody.content());
         thread.addForumPost(post);
         creator.getForumPostList().add(post);
         this.postRepository.save(post);
         return post;
     }
+
+    public ForumThread toggleThreadFollowing(Integer threadId, Integer userId){
+        final ForumThread thread = this.forumRepository.findById(threadId).orElseThrow();
+        final User follower = this.userRepository.findById(userId).orElseThrow();
+
+        if(follower.isFollowing(thread)){
+            thread.getFollowingUsers().remove(follower);
+            follower.getFollowedThreads().remove(thread);
+        }else{
+            thread.getFollowingUsers().add(follower);
+            follower.getFollowedThreads().add(thread);
+        }
+        return thread;
+    }
```

Comment:
A to będzie działało? Bo nie widzę, żeby `follower` i `thread` były updatowane, trzeba najwyżej dodać `repository.save` na obydwu